app/src/main/java/com/greenaddress/greenbits/ui/MainFragment.java

```diff
@@ -136,6 +139,17 @@ public void onClick(final View v) {
             reloadTransactions(false, true);
         }
 
+        mSwipeRefreshLayout = UI.find(mView, R.id.mainTransactionListSwipe);
+        mSwipeRefreshLayout.setColorSchemeColors(ContextCompat.getColor(getContext(), R.color.accent));
+        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                Log.d(TAG, "onRefresh -> " + TAG);
+                updateBalance();
+                reloadTransactions(false, false);
```

Comment:
You can probably just call onBalanceUpdated() here no? (with a comment stating this is used to cause a refresh)